Decompress JPEG 2000 images with  YCC 422 or YCC 420 sub-sampling on the 3 channels fails with: "OpenJPEG2000: tiles are not supported"

```
$ dcm2jpg /tmp/TCGA-AF-2687-01Z-00-DX1_im_0-1.dcm /tmp/frame1.jpg
Native OpenCV library has been found in /home/gunter/dcm4che/lib/linux-x86-64/libopencv_java.so
11:50:14.256 DEBUG - Decompressor: org.dcm4che3.opencv.NativeImageReader
11:50:14.258 DEBUG - Start decompressing frame #1
Mar 04, 2024 11:50:14 AM org.opencv.osgi.OpenCVNativeLoader init
INFO: Successfully loaded OpenCV native library.
imread_('memory image'): can't read data: OpenCV(4.8.1) /work/modules/imgcodecs/src/grfmt_jpeg2000_openjpeg.cpp:689: error: (-2:Unspecified error) in function 'virtual bool cv::detail::Jpeg2KOpjDecoderBase::readData(cv::Mat&)'
> OpenJPEG2000: tiles are not supported (expected: '(int)comp.dx == 1'), where
>     '(int)comp.dx' is 2
> must be equal to
>     '1' is 1

Failed to convert /tmp/TCGA-AF-2687-01Z-00-DX1_im_0-1.dcm: w and h must be > 0
java.lang.IllegalArgumentException: w and h must be > 0
	at java.desktop/java.awt.image.Raster.createInterleavedRaster(Raster.java:281)
	at org.weasis.opencv.op.ImageConversion.toBufferedImage(ImageConversion.java:84)
	at org.weasis.opencv.op.ImageConversion.toBufferedImage(ImageConversion.java:120)
	at org.dcm4che3.opencv.NativeImageReader.read(NativeImageReader.java:252)
	at org.dcm4che3.imageio.plugins.dcm.DicomImageReader.read(DicomImageReader.java:458)
	at org.dcm4che3.tool.dcm2jpg.Dcm2Jpg.readImageFromDicomInputStream(Dcm2Jpg.java:438)
	at org.dcm4che3.tool.dcm2jpg.Dcm2Jpg.convert(Dcm2Jpg.java:425)
	at org.dcm4che3.tool.dcm2jpg.Dcm2Jpg.mconvert(Dcm2Jpg.java:412)
	at org.dcm4che3.tool.dcm2jpg.Dcm2Jpg.main(Dcm2Jpg.java:381)
```
[TCGA-AF-2687-01Z-00-DX1_im_0-1.dcm.gz](https://github.com/dcm4che/dcm4che/files/14480551/TCGA-AF-2687-01Z-00-DX1_im_0-1.dcm.gz)

TCGA-AF-2687-01Z-00-DX1_im_0-1.dcm encapsulates the first frame of VL Whole Slide Microscopy Image: `IDC/TCGA-AF-2687/images/TCGA-AF-2687-01Z-00-DX1_im_0.dcm` in [test data](https://www.dropbox.com/scl/fo/wn6qfm8rc5eyojj6rfj27/h?rlkey=r8pb5bwj95xia1120mf5vr263&dl=0) for upcoming WG26 Annotation Connectathon.

Considering
```
1248: (0008,1090) LO #52 [Aperio converted by com.pixelmed.convert.TIFFToDicom] ManufacturerModelName
```
and
```
1844: (0018,A001) SQ #-1 ContributingEquipmentSequence
1856: >(FFFE,E000) #-1 Item #1
1864: >(0008,0070) LO #8 [PixelMed] Manufacturer
1880: >(0008,0080) LO #8 [PixelMed] InstitutionName
1896: >(0008,0081) ST #10 [Bangor, PA] InstitutionAddress
1914: >(0008,1040) LO #20 [Software Development] InstitutionalDepartmentName
1942: >(0008,1090) LO #32 [com.pixelmed.convert.TIFFToDicom] ManufacturerModelName
1982: >(0018,1020) LO #34 [Vers. Sat May 28 11:42:06 EDT 2022] SoftwareVersions
2024: >(0018,A002) DT #24 [20220612165335.074+0000] ContributionDateTime
2056: >(0018,A003) ST #24 [TIFF to DICOM conversion] ContributionDescription
```
of the VL Whole Slide Microscopy Image, the JPEG2000 object may have been generated by https://github.com/ImagingDataCommons/idc-wsi-conversion

Verified, that the failure occurs also with plain OpenCV 4.5.4:

```
$ python3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cv2
>>> print(cv2.__version__)
4.5.4
>>> import sys
>>> img = cv2.imread("/tmp/TCGA-AF-2687-01Z-00-DX1_im_0-1.j2k")
[ WARN:0] global ./modules/imgcodecs/src/grfmt_jpeg2000_openjpeg.cpp (607) readData OpenJPEG2000: Image has unknown or unspecified color space, SRGB is assumed
imread_('/tmp/TCGA-AF-2687-01Z-00-DX1_im_0-1.j2k'): can't read data: OpenCV(4.5.4) ./modules/imgcodecs/src/grfmt_jpeg2000_openjpeg.cpp:641: error: (-2:Unspecified error) in function 'virtual bool cv::detail::Jpeg2KOpjDecoderBase::readData(cv::Mat&)'
> OpenJPEG2000: tiles are not supported (expected: '(int)comp.dx == 1'), where
>     '(int)comp.dx' is 2
> must be equal to
>     '1' is 1
```

but not with plain OpenJPEG v2.4.0:
```console
$ opj_decompress -i /tmp/TCGA-AF-2687-01Z-00-DX1_im_0-1.j2k -o /tmp/TCGA-AF-2687-01Z-00-DX1_im_0-1.png

[INFO] Start to read j2k main header (0).
[INFO] Main header has been correctly decoded.
[INFO] No decoded area parameters, set the decoded area to the whole image
[INFO] Header of tile 1 / 1 has been read.
[INFO] Generated Outfile /tmp/TCGA-AF-2687-01Z-00-DX1_im_0-1.png
decode time: 2 ms
```

--- a/README.md
+++ b/README.md
@@ -17,10 +17,8 @@ dcm4che uses a native library for the compression and decompression of images. H
 | System  | Architecture | Package        | Requirement            |
 |:--------|:-------------|:---------------|:-----------------------|
 | Linux   | x86 64-bit   | linux-x86-64   | GLIBC_2.17             |
-| Linux   | x86 32-bit   | linux-x86      | GLIBC_2.17             |
 | Linux   | ARM 64-bit   | linux-aarch64  | GLIBC_2.27             |
-| Linux   | ARM 32-bit   | linux-armv7a   | GLIBC_2.17             |
-| Linux   | s390x        | linux-s390x    | GLIBC_2.17             |
+| Linux   | ARM 32-bit   | linux-armv7a   | GLIBC_2.28             |
 | Windows | x86 64-bit   | windows-x86-64 | Windows 7 or higher    |
 | Windows | x86 32-bit   | windows-x86    | Windows 7 or higher    |
 | Mac OS  | x86 64-bit   | macosx-x86-64  | Mac OS 10.13 or higher |

--- a/dcm4che-assembly/pom.xml
+++ b/dcm4che-assembly/pom.xml
@@ -128,6 +128,48 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-antrun-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>exportAntProperties</id>
+            <goals>
+              <goal>run</goal>
+            </goals>
+            <phase>validate</phase>
+            <configuration>
+              <target>
+                <!-- <echoproperties />-->
+                <!-- <echo>some.libopencv_java.path=${org.weasis.thirdparty.org.opencv:libopencv_java:so:linux-x86-64}</echo>-->
+              </target>
+              <exportAntProperties>true</exportAntProperties>
+              <!-- @NOTE : exportAntProperties is required for build-helper-maven to read the supplied value property ${groupId:artifactId:type[:classifier]}-->
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>resolve weasis.opencv.native.version - property</id>
+            <phase>validate</phase>
+            <goals>
+              <goal>regex-property</goal>
+            </goals>
+            <configuration>
+              <name>weasis.opencv.native.version</name>
+              <!--suppress UnresolvedMavenProperty -->
+              <value>${org.weasis.thirdparty.org.opencv:libopencv_java:so:linux-x86-64}</value>
+              <regex>.*libopencv_java\/(.*)\/libopencv_java.*</regex>
+              <replacement>$1</replacement>
+              <failIfNoMatch>true</failIfNoMatch>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
     </plugins>
   </build>
   <dependencies>
@@ -568,63 +610,42 @@
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>so</type>
       <classifier>linux-x86-64</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
-      <type>so</type>
-      <classifier>linux-x86</classifier>
-    </dependency>
-    <dependency>
-      <groupId>org.weasis.thirdparty.org.opencv</groupId>
-      <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
-      <type>so</type>
-      <classifier>linux-s390x</classifier>
-    </dependency>
-    <dependency>
-      <groupId>org.weasis.thirdparty.org.opencv</groupId>
-      <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>so</type>
       <classifier>linux-armv7a</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>so</type>
       <classifier>linux-aarch64</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>dylib</type>
       <classifier>macosx-x86-64</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>dylib</type>
       <classifier>macosx-aarch64</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>opencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>dll</type>
       <classifier>windows-x86</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>opencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>dll</type>
       <classifier>windows-x86-64</classifier>
     </dependency>

--- a/dcm4che-assembly/src/main/assembly/component.xml
+++ b/dcm4che-assembly/src/main/assembly/component.xml
@@ -242,22 +242,6 @@
     </dependencySet>
     <dependencySet>
       <outputFileNameMapping>${artifact.artifactId}.${artifact.extension}</outputFileNameMapping>
-      <outputDirectory>lib/linux-x86</outputDirectory>
-      <includes>
-        <include>*:*:so:linux-x86:*</include>
-      </includes>
-      <useProjectArtifact>false</useProjectArtifact>
-    </dependencySet>
-    <dependencySet>
-      <outputFileNameMapping>${artifact.artifactId}.${artifact.extension}</outputFileNameMapping>
-      <outputDirectory>lib/linux-s390x</outputDirectory>
-      <includes>
-        <include>*:*:so:linux-s390x:*</include>
-      </includes>
-      <useProjectArtifact>false</useProjectArtifact>
-    </dependencySet>
-    <dependencySet>
-      <outputFileNameMapping>${artifact.artifactId}.${artifact.extension}</outputFileNameMapping>
       <outputDirectory>lib/linux-armv7a</outputDirectory>
       <includes>
         <include>*:*:so:linux-armv7a:*</include>

--- a/dcm4che-imageio-opencv/pom.xml
+++ b/dcm4che-imageio-opencv/pom.xml
@@ -17,7 +17,6 @@
     <dependency>
       <groupId>org.weasis.core</groupId>
       <artifactId>weasis-core-img</artifactId>
-      <version>${weasis.core.img.version}</version>
     </dependency>
   </dependencies>
 </project>

--- a/dcm4che-imageio-test/pom.xml
+++ b/dcm4che-imageio-test/pom.xml
@@ -241,7 +241,6 @@
                 <artifactItem>
                   <groupId>org.weasis.thirdparty.org.opencv</groupId>
                   <artifactId>${lib-file-name}</artifactId>
-                  <version>${weasis.opencv.native.version}</version>
                   <type>${lib-file-ext}</type>
                   <classifier>${os-name}-${cpu-name}</classifier>
                   <overWrite>false</overWrite>

--- a/dcm4che-jboss-modules/pom.xml
+++ b/dcm4che-jboss-modules/pom.xml
@@ -162,61 +162,46 @@
     <dependency>
       <groupId>org.weasis.core</groupId>
       <artifactId>weasis-core-img</artifactId>
-      <version>${weasis.core.img.version}</version>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>so</type>
       <classifier>linux-x86-64</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
-      <type>so</type>
-      <classifier>linux-x86</classifier>
-    </dependency>
-    <dependency>
-      <groupId>org.weasis.thirdparty.org.opencv</groupId>
-      <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>so</type>
       <classifier>linux-armv7a</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>so</type>
       <classifier>linux-aarch64</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>dylib</type>
       <classifier>macosx-x86-64</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>libopencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>dylib</type>
       <classifier>macosx-aarch64</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>opencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>dll</type>
       <classifier>windows-x86</classifier>
     </dependency>
     <dependency>
       <groupId>org.weasis.thirdparty.org.opencv</groupId>
       <artifactId>opencv_java</artifactId>
-      <version>${weasis.opencv.native.version}</version>
       <type>dll</type>
       <classifier>windows-x86-64</classifier>
     </dependency>

--- a/dcm4che-jboss-modules/src/main/assembly/modules.xml
+++ b/dcm4che-jboss-modules/src/main/assembly/modules.xml
@@ -155,14 +155,6 @@
     </dependencySet>
     <dependencySet>
       <outputFileNameMapping>${artifact.artifactId}.${artifact.extension}</outputFileNameMapping>
-      <outputDirectory>/modules/org/dcm4che/imageio/main/lib/linux-i686</outputDirectory>
-      <includes>
-        <include>*:*:so:linux-x86:*</include>
-      </includes>
-      <useProjectArtifact>false</useProjectArtifact>
-    </dependencySet>
-    <dependencySet>
-      <outputFileNameMapping>${artifact.artifactId}.${artifact.extension}</outputFileNameMapping>
       <outputDirectory>/modules/org/dcm4che/imageio/main/lib/linux-x86_64</outputDirectory>
       <includes>
         <include>*:*:so:linux-x86-64:*</include>

--- a/pom.xml
+++ b/pom.xml
@@ -81,8 +81,7 @@
     <maven.compiler.target>1.8</maven.compiler.target>
     <slf4j.version>2.0.9</slf4j.version>
     <logback.version>1.4.14</logback.version>
-    <weasis.core.img.version>4.8.1</weasis.core.img.version>
-    <weasis.opencv.native.version>4.8.1-dcm</weasis.opencv.native.version>
+    <weasis.core.img.version>4.9.0.1</weasis.core.img.version>
     <keycloak.version>24.0.1</keycloak.version>
     <jbossws-cxf-client.version>7.0.0.Final</jbossws-cxf-client.version>
     <apache-cxf.version>4.0.0</apache-cxf.version>
@@ -333,6 +332,18 @@
         <groupId>com.sun.xml.ws</groupId>
         <artifactId>rt</artifactId>
         <version>${jakarta.xml.ws.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.weasis.core</groupId>
+        <artifactId>weasis-core-img-bom</artifactId>
+        <version>${weasis.core.img.version}</version>
+        <scope>import</scope>
+        <type>pom</type>
+      </dependency>
+      <dependency>
+        <groupId>org.weasis.core</groupId>
+        <artifactId>weasis-core-img</artifactId>
+        <version>${weasis.core.img.version}</version>
       </dependency>
     </dependencies>
   </dependencyManagement>

